renku/core/commands/workflow.py

```diff
@@ -48,98 +49,69 @@ def _deref(ref):
+    if isinstance(workflow, CompositePlan):
+        return CompositePlanViewModel.from_composite_plan(workflow)
+    return PlanViewModel.from_plan(workflow)
 
 
 @inject.autoparams()
-def _set_workflow_name(name, path, force, client: LocalClient):
-    """Sets the <name> for remote <path>."""
-    from renku.core.models.refs import LinkReference
-
-    LinkReference.create(client=client, name=_ref(name), force=force).set_reference(path)
+def _find_workflow(name_or_id: str, database: Database) -> Union[CompositePlan, Plan]:
+    workflow = database["plans"].get(name_or_id)
 
+    if not workflow:
+        workflow = database["plans-by-name"].get(name_or_id)
 
-def set_workflow_name_command():
-    """Command that sets the <name> for remote <path>."""
-    return Command().command(_set_workflow_name).require_clean().with_commit()
+    if not workflow:
+        raise errors.ParameterError(f'The specified workflow "{name_or_id}" cannot be found.')
+    return workflow
 
 
 @inject.autoparams()
-def _rename_workflow(old, new, force, client: LocalClient):
-    """Rename the workflow named <old> to <new>."""
-    from renku.core.models.refs import LinkReference
-
-    LinkReference(client=client, name=_ref(old)).rename(_ref(new), force=force)
-
-
-def rename_workflow_command():
-    """Command that renames the workflow named <old> to <new>."""
-    return Command().command(_rename_workflow).require_clean().with_commit()
-
+def _list_workflows(database: Database, format: str, columns: List[str]):
+    """List or manage workflows with subcommands."""
+    workflows = unique_workflow(database)
 
-@inject.autoparams()
-def _remove_workflow(name, client: LocalClient):
-    """Remove the remote named <name>."""
-    from renku.core.models.refs import LinkReference
+    if format not in WORKFLOW_FORMATS:
+        raise errors.UsageError(f'Provided format "{format}" is not supported ({", ".join(WORKFLOW_FORMATS.keys())})"')
 
-    LinkReference(client=client, name=_ref(name)).delete()
+    return WORKFLOW_FORMATS[format](list(map(lambda x: _to_view(x), workflows.values())), columns=columns)
 
 
-def remove_workflow_command():
-    """Command that removes the remote named <name>."""
-    return Command().command(_remove_workflow).require_clean().with_commit()
+def list_workflows_command():
+    """Command to list or manage workflows with subcommands."""
+    return Command().command(_list_workflows).require_migration().with_database(write=False)
 
 
 @inject.autoparams()
-def _create_workflow(output_file, revision, paths, client: LocalClient):
-    """Create a workflow description for a file."""
-    graph = Graph()
-    outputs = graph.build(paths=paths, revision=revision)
-
-    workflow = graph.as_workflow(outputs=outputs)
-
-    if output_file:
-        output_file = Path(output_file)
+def _remove_workflow(name, force, database: Database):
+    """Remove the workflow named <name>."""
+    workflows = unique_workflow(database)
```

Comment:
unique_workflow won't be needed since the Gateways take care of it.

but the Gateway probably needs to be extended to deal with deleted workflows.